### PR TITLE
Use 'glob-gitignore' instead of 'glob'

### PR DESCRIPTION
**Pull request by evidolob**
This PR enhance `alizer` npm package:

- Use `glob-gitignore` instead of `glob` to list target files, by default it will ignore file/folder listed in `.gitignore` file
- Add `build` npm script (`clean` + `compile`)
- Add new line and end of `*.ts` files

### CODE REVIEW

**Comment by evidolob** [comment]
cc @jeffmaury @lstocchi

File: js/src/recognizer.ts [hunk 1]
```diff
@@ -75,13 +74,6 @@ async function getDetailedLanguage(language: string, files: string[]): Promise<L
...
-        glob(`${root}/**/*.*`, `!${root}/**`, (err: any, files: string[] | PromiseLike<string[]>) => {
-            if (err) {
-                reject(`Unable to retrieve files from current project ${root}. ${err}`);
-            }
-            return resolve(files);
-        })
-    });
-}
\ No newline at end of file
+function getFiles(root: string): Promise<string[]> {
```
Comment (lstocchi) [inline js/src/recognizer.ts:77]: Shouldn't this be async?

**Review by lstocchi — COMMENTED** [review]
I tried it but it looks like the glob-ignore library doesn't work as expected. If i add a rule (to ignore .java files) inside the .gitignore in the quarkus project used for tests, it still fetches all .java files. Am i missing something?

File: js/src/recognizer.ts (hunk 1 above)
Comment (evidolob) [reply js/src/recognizer.ts:77]: No, if we not call `await` inside function body

**Comment by evidolob** [comment]
Hm, that strange, let me check

File: js/src/recognizer.ts (hunk 1 above)
Comment (lstocchi) [reply js/src/recognizer.ts:77]: Right now it is, look at line 22

**Comment by evidolob** [comment]
@lstocchi It turndown that `glob-gitignore` is only supports of `.gitignore` rules but doesn't read it automatically, I just  updated PR with reading   `.gitignore` and pass it to  `glob-gitignore` when we perform search.

**Review by lstocchi — APPROVED** [review]
LGTM

**Review by jeffmaury — APPROVED** [review]
This is a partial implementation of .gitignore as it does only handle the root .gitignore

Maybe worth having .gitignore parsing and loading in a separate function

Unit test are welcomed

**Pull request merged**